### PR TITLE
ENH: Add AE title option to DIMSE export

### PR DESCRIPTION
**Pull request by riep**
AE title is currently hard-coded, add a text edit in the dicom export dialog.

### CODE REVIEW

**Review by cpinter — DISMISSED** [review]
LGTM

File: Modules/Scripted/DICOMLib/DICOMSendDialog.py
Comment (lassoan) [inline Modules/Scripted/DICOMLib/DICOMSendDialog.py:47]:
```suggestion
    self.serverAETitleEdit.text = self.settings.value('DICOM/Send/AETitle', 'CTK')
```

File: Modules/Scripted/DICOMLib/DICOMSendDialog.py
Comment (lassoan) [inline Modules/Scripted/DICOMLib/DICOMSendDialog.py:82]:
```suggestion
    self.settings.setValue('DICOM/Send/AETitle', aeTitle)
```

File: Modules/Scripted/DICOMLib/DICOMSendDialog.py
Comment (lassoan) [inline Modules/Scripted/DICOMLib/DICOMSendDialog.py:105]: Do not keep adding and removing this connection (it is unexpected and very error-prone). Add it once, when the widget is set up.

**Review by lassoan — COMMENTED** [review]
Thank you for working on this. At some point we need to add a proper host list editor for DICOM sending (merged with the host editor of Query/Retrieve), but until then this is a good solution.

I've just added a few comments that should be very easy to fix.

File: Modules/Scripted/DICOMLib/DICOMSendDialog.py
Comment (riep) [reply Modules/Scripted/DICOMLib/DICOMSendDialog.py]: done

File: Modules/Scripted/DICOMLib/DICOMSendDialog.py
Comment (riep) [reply Modules/Scripted/DICOMLib/DICOMSendDialog.py]: done

File: Modules/Scripted/DICOMLib/DICOMSendDialog.py
Comment (riep) [reply Modules/Scripted/DICOMLib/DICOMSendDialog.py]: done

File: Modules/Scripted/DICOMLib/DICOMSendDialog.py
Comment (riep) [inline Modules/Scripted/DICOMLib/DICOMSendDialog.py:48]: **T**ite to be consistent

File: Modules/Scripted/DICOMLib/DICOMProcesses.py
Comment (lassoan) [inline Modules/Scripted/DICOMLib/DICOMProcesses.py:351]: I've just noticed that this change would break those extensions that use DICOMSender class. An easy change would be to add a default None value to aeTitle (and it that case use the default "CTK" value), and maybe move it to the end of the argument list.

Comment (riep) [reply Modules/Scripted/DICOMLib/DICOMProcesses.py:351]: Sure I do it

**Comment by riep** [comment]
sorry @cpinter "dismissed cpinter’s stale review via d0ccf19 1 hour ago" was a miss click

**Comment by jamesobutler** [comment]
Hi @riep, could you run Utilities/SetupForDevelopment.sh? This will configure some user details allowing for your GitHub details to be clicked on next to the individual commit. 
image
Then force push the commit again to try and updated the details for this commit.

**Comment by riep** [comment]
Hi James,
I have done it but, it seems that it is not working. I inverstigate that.

**Comment by riep** [comment]
Now the two are present ...

**Comment by riep** [comment]
@lassoan @jamesobutler  please let me know if additional things must be done before merging this PR.

File: Modules/Scripted/DICOMLib/DICOMSendDialog.py
Comment (lassoan) [inline Modules/Scripted/DICOMLib/DICOMSendDialog.py:50]:
```suggestion
    self.serverAETitleEdit.enabled = self.protocolSelectorCombobox.currentText == 'DIMSE'
```

It is better to avoid "magic numbers". Comparing displayed text is not very robust (when we add translation then we'll need to switch to using item UserData), but for now it should be good.

File: Modules/Scripted/DICOMLib/DICOMSendDialog.py
Comment (lassoan) [inline Modules/Scripted/DICOMLib/DICOMSendDialog.py:75]:
```suggestion
    self.serverAETitleEdit.enabled = self.protocolSelectorCombobox.currentText == 'DIMSE'
```

**Review by lassoan — COMMENTED** [review]
Thank you, it looks good, I just added two minor suggestions for making the code a bit more readable. You should be able to just click to "commit the suggestion" on the web GUI (no need to change locally on your computer and push to github).

**Comment by riep** [comment]
Done thank you Andreas

**Comment by lassoan** [comment]
Merged. Thank you for your contribution!

**Pull request merged**
